crates/runtime/src/dataconnector/listing/connector.rs

```diff
@@ -684,6 +697,65 @@ async fn get_last_modified(
     }
 }
 
+async fn verify_schema_source_path(
```

Comment:
Alternative approach would be to re-use existing `get_last_modified` fn and pass `schema_source_path`, the only difference that we would like to exit early (no need to list everything), plus limit files to scan. Based on `schema_source_path` purpose I don't expect schema_source_path to point to the location with too many objects